File: CS/LegacyCommands/SimpleCommandUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace DxSampleLegacyCommands {
    public partial class SimpleCommandUserControl : XtraUserControl {
        public SimpleCommandUserControl() {
            InitializeComponent();
            #region SetUp
            SimpleButton commandButton = new SimpleButton();
            commandButton.Text = "Execute Command";
            commandButton.Dock = DockStyle.Top;
            commandButton.Parent = this;
            #endregion SetUp

            #region #simpleCommand
            // This is simple legacy-command. It provide the Execute method for doing something.
            LegacyCommand command = new LegacyCommand();
            // UI binding for button
            commandButton.BindCommand(command);
            #endregion #simpleCommand
        }
        #region CleanUp
        void OnDisposing() {
        }
        #endregion CleanUp
        public class LegacyCommand {
            public void Execute(object parameter) {
                XtraMessageBox.Show("Hello! I'm  Legacy command and I'm running!");
            }
        }
    }
}

File: VB/LegacyCommands/SimpleCommandUserControl.vb
Imports DevExpress.MVVM
Imports DevExpress.MVVM.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace DxSampleLegacyCommands
    Partial Public Class SimpleCommandUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim commandButton As New SimpleButton()
            commandButton.Text = "Execute Command"
            commandButton.Dock = DockStyle.Top
            commandButton.Parent = Me

            ' This is simple legacy-command. It provide the Execute method for doing something.
            Dim command As New LegacyCommand()
            ' UI binding for button
            commandButton.BindCommand(command)
        End Sub
         Private Sub OnDisposing()
        End Sub
        Public Class LegacyCommand
            Public Sub Execute(ByVal parameter As Object)
                XtraMessageBox.Show("Hello! I'm  Legacy command and I'm running!")
            End Sub
        End Class
    	End Class
End Namespace